go/rows.go

```diff
@@ -125,6 +129,16 @@ func (r *Rows) ColumnTypeDatabaseTypeName(index int) string {
 
 // Next is to get next result set page.
 func (r *Rows) Next(dest []driver.Value) error {
+	var err error
+	r.resultsOpened.Do(func() {
+		// download the results and open the resulting CSV for reading
+		err = r.openResults()
+	})
+	if err != nil {
+		r.reachedLastPage = true
+		return err
+	}
+
 	if r.reachedLastPage {
```

Comment:
this if condition could be combined with the above.

Reply:
It is possible for `reachedLastPage` to be set elsewhere (it's effectively `closed`) and `err` would be `nil` after the first call to `Next`. Not the clearest, but I think it does what we want.